### PR TITLE
Cluster failover fixes

### PR DESCRIPTION
**Pull request by rajesh-blueshift**
Handle redis cluster failover conditions gracefully

* If there no nodes for a slot, we try hitting a random node to get us redirected to a correct node
* This error was happening in the `get_connection_by_slot` function with `@slots[slot]` not having any key for the slot - which is not possible in a regular scenario

### CODE REVIEW

File: rediscluster.rb [hunk 1]
```diff
@@ -191,7 +191,19 @@ def send_cluster_command(argv, master_only: true, &blk)
...
+        # there is a chance that some slots are unassigned.
+        # In this situation, we should try connecting to a random node.
+        # That will either trigger a "MOVED" or "ASK" response
+        # and hopefully we will reach the appropriate node within the limited iterations
+        r =
+          if @connections.slot_has_nodes?(slot)
+            @connections.get_connection_by_slot(slot, master_only)
+          else
+            puts "Cannot find nodes for slot #{slot}, trying with a random node"
+            sleep(@cluster_retry_interval) if ttl < RedisClusterRequestTTL/2
```
Comment (rajesh-blueshift) [inline rediscluster.rb:203]: I have kept the sleep conditional similar to the pattern adopted in rest of the file

Comment (cibingeorge) [reply rediscluster.rb:203]: you have puts on it

File: rediscluster.rb [hunk 2]
```diff
@@ -204,6 +216,7 @@ def send_cluster_command(argv, master_only: true, &blk)
         sleep(@cluster_retry_interval) if ttl < RedisClusterRequestTTL/2
       rescue => e
         errv = e.to_s.split
+        puts "Received response=#{errv}"
```
Comment (cibingeorge) [inline rediscluster.rb]: puts

**Review by chongbomb — COMMENTED** [review]
@rajesh-blueshift This can be approved once you remove the puts as Cibin mentioned.

File: rediscluster.rb (hunk 2 above)
Comment (rajesh-blueshift) [reply rediscluster.rb]: removed now

File: rediscluster.rb (hunk 1 above)
Comment (rajesh-blueshift) [reply rediscluster.rb:203]: removed now

**Comment by rajesh-blueshift** [comment]
> @rajesh-blueshift This can be approved once you remove the puts as Cibin mentioned.

I have removed it now @chongbomb

**Review by chongbomb — APPROVED** [review]

**Pull request merged**